### ISSUE TITLE
CW-426: Make "View solution" easier to use

### ISSUE THREAD

**Issue by hobovsky** [Codewars/codewars.com#2828]
Initially asked on [Discord](https://discord.com/channels/846624424199061524/846624424199061527/1024381045174440116):

In kata discourse, under user posts, there's this "View solution" button which reveals last attempted user solution, even if they did not solve the kata. It turned out to be a great help when supporting users who complain (correctly or not) on wrong tests, or ask for help. Unfortunately, this function went unnoticed by me for for a really long time, because, frankly, I did not know how it works and I just thought it's broken. It's a real pain to use correctly, and requires a lot of hassle:
- I have to find out what language the user attempted the kata with. Currently I use mods power and "solution stats" page to check user's favorite language (as pointed out by B4B I could use "Last Trained" information form user profile, and it works in most cases, but sometimes users change the language after encountering a problem with tests and this information is not helpful anymore).
- I have to solve the kata in this language.
- I have to switch the language dropdown to the language, which is again somewhat annoying due to loading times for large discourses.

I am curious if it would be possible to improve the experience of users who try to help out others. My idea would be to make the section independent from currently selected language, and show information about all languages attempted by a user: show code for languages I also solved the kata with, and just some small marker or "this solution is locked" for languages I have not solved.
Additionally, since mods can already see all solutions of users, maybe show all solutions of a user when viewed by mods.

**Comment by hobovsky** [comment]
[Answered](https://discord.com/channels/846624424199061524/846624424199061527/1024463050851504198) by @kazk:

> [...] I probably won't have the time for complex changes like you suggested anytime soon, but I've been thinking of automatically tagging a new comment with the user's active language. That should help a lot with relatively low effort.

**Comment by hobovsky** [comment]
Bonus points for making the solution easy to train (by forking, or opening the trainer with the code already in the editor, ideally in a new tab).

**Comment by benjaminzwhite** [comment]
Related to hobovsky's comment:

I've tried to help out several times where a user has posted a generic "it doesn't work message", so I've copied their code into my IDE, solved the kata to unlock tests, tried testing their solution, only to find that it seems to work fine.  

Then I reply to them, trying to close the Issue that they have invariably raised, and it turns out that they managed to solve their problem but just don't have the etiquette/good manners to return to their discourse page and close their own comment.

TL;DR - would it be possible to flag a user's solution as having passed the kata? Just a simple green checkmark for example, so menders could automatically close dozens of outstanding queries where users eventually figured out themselves what was going wrong with their code?

**Comment by VanessaGreene** [comment]
Hi all, specifically @hobovsky; thank you for re-raising this issue with me. I'm happy to confirm that this has been escalated and is now being scoped by the team. I hope to update you on our next steps within 2 weeks.

**Comment by VanessaGreene** [comment]
Hi @hobovsky, this issue has been assigned and added to our up-next list. We hope to have an update shortly.

**Comment by hobovsky** [comment]
Not gonna lie, this is one of the most anticipated features by me, and I believe by many othe users helping newbies to find problems with their solutions. Can't wait!
